tests/metrics/test_stratification.py

```diff
@@ -1,371 +1,252 @@
+import itertools
 from datetime import datetime
 from math import floor
 from typing import Callable, List, Set
 from unittest.mock import MagicMock, patch
 
+import numpy as np
 import pandas as pd
 import pytest
+from vivarium import InteractiveContext
+from vivarium.framework.engine import Builder
+from vivarium.framework.population import SimulantData
+from vivarium.testing_utilities import TestPopulation, metadata
 
 from vivarium_public_health.metrics.stratification import (
-    ResultsStratifier,
+    ResultsStratifier as ResultsStratifier_,
+)
+from vivarium_public_health.metrics.stratification import (
     Source,
     SourceType,
     StratificationLevel,
 )
 
-AGE_BINS = pd.DataFrame(
-    [
-        {"age_start": age, "age_end": age + 5, "age_group_name": f"{age}_to_{age + 5}"}
-        for i, age in enumerate(range(10, 65, 5))
-    ]
-)
-DEFAULT_CONFIGS = ["year", "sex", "age"]
-START_YEAR = 2020
-END_YEAR = 2030
-STRATIFICATION_LEVELS = {
-    "age": StratificationLevel(
-        name="age",
-        sources=[Source("age", SourceType.COLUMN)],
-        categories={"0", "1", "2", "3", "4"},
-        mapper=lambda row: str(floor(row[0])),
-    ),
-    "sex": StratificationLevel(
-        name="l2",
-        sources=[Source("sex", SourceType.COLUMN)],
-        categories={"Male", "Female"},
-    ),
-    "multi": StratificationLevel(
-        name="multi",
-        sources=[
-            Source("part_1", SourceType.COLUMN),
-            Source("pipeline_1", SourceType.PIPELINE),
-        ],
-        categories={"c2", "c30", "c31", "c10", "c11"},
-        mapper=lambda row: "c2"
-        if row["part_1"] == "c2"
-        else row["part_1"] + row["pipeline_1"],
-    ),
-    "p2": StratificationLevel(
-        name="p2",
-        sources=[Source("pipeline_2", SourceType.PIPELINE)],
-        categories={"2", "3"},
-    ),
-    "year": StratificationLevel(
-        name="year",
-        sources=[Source("clock_1", SourceType.CLOCK)],
-        categories={"2020", "2021", "2022", "2023"},
-        mapper=lambda row: str(row[0].year),
-        current_categories_getter=lambda: {str(mock_clock().year)},
-    ),
-    "month": StratificationLevel(
-        name="month",
-        sources=[Source("clock_2", SourceType.CLOCK)],
-        categories={str(i + 1) for i in range(12)},
-        mapper=lambda row: str(row[0].month),
-        current_categories_getter=lambda: {str(mock_clock().month)},
-    ),
-}
 
+class FavoriteColor:
```

Comment:
I'm also shifting strategies.  Rajan's test were all "white box" tests which had intimate knowledge of the stratifier component internals.  What I'm doing instead is "black box" testing, which doesn't know anything about the component beyond it's public interface and contracts. I'm doing this by generating minimal interactive simulations with all the features necessary to stress test the results stratifier and then testing the component interface.  This embeds the new component in a stripped down, but realistic setting where it's much easier to check for expected behavior.